R/ResultFlagsIndependent.R

```diff
@@ -1254,3 +1254,307 @@ TADA_FindPotentialDuplicatesSingleOrg <- function(.data) {
+#' and TADA.ResultValueAggregation.DepthProfile.Flag. If a daily_agg = "ave",
+#' "min", or "max", aggregated values will be identified in the TADA.ResultAggregation.Flag
+#' column. In the case of daily_agg = "ave", additional rows to display averages will be
+#' added to the data frame. They can be identified by the prefix ("TADA-") of
+#' their result identifiers.
+#'
+#' @export
+#'
+#' @examples
+#' # Load dataset
+#' data(Data_6Tribes_5y)
+#'
+#' # assign TADA.DepthCategory.Flag with no aggregation
+#' # Data_6Tribs_5y_DepthCat <- TADA_DepthCategory.Flag(Data_6Tribes_5y)
+#'
+#' # assign TADA.DepthCategory.Flag and determine average values by depth category
+#' Data_6Tribs_5y_Mean <- TADA_DepthCategory.Flag(Data_6Tribes_5y, bycategory = TRUE, daily_agg = "ave")
+#'
+TADA_DepthCategory.Flag <- function(.data, bycategory = FALSE, daily_agg = "no", clean = FALSE) {
+  depthcat.list <- c("Epilimnion-surface", "Hypolimnion-bottom", "Metalimnion/Thermocline-middle")
+
+  ard.ref <- utils::read.csv(system.file("extdata", "WQXActivityRelativeDepthRef.csv", package = "TADA")) %>%
+    dplyr::rename(
+      ARD_Category = TADA.DepthCategory.Flag,
+      ActivityRelativeDepthName = Name
+    ) %>%
+    dplyr::select(ARD_Category, ActivityRelativeDepthName)
+
+  depth.count <- .data %>%
+    dplyr::filter(!is.na(TADA.ActivityDepthHeightMeasure.MeasureValue) |
+      !is.na(TADA.ResultDepthHeightMeasure.MeasureValue)) %>%
+    nrow()
+
+  if (depth.count > 0) {
+    print(paste("TADA_DepthCategory.Flag: checking data set for depth values. ", depth.count, " results have depth values available.", sep = ""))
+
+    print("TADA_DepthCategory.Flag: assigning depth categories.")
+
+    .data <- .data %>%
+      # set equal to TADA.ResultDepthHeighMeasure.MeasureValue if available, otherwise use TADA.ActivityDepthHeightMeasure.MeasureValue
+      dplyr::mutate(Depth = ifelse(!is.na(TADA.ResultDepthHeightMeasure.MeasureValue), TADA.ResultDepthHeightMeasure.MeasureValue,
+        TADA.ActivityDepthHeightMeasure.MeasureValue
+      )) %>%
+      # use group_by to identify profile data
+      dplyr::group_by(ActivityStartDate, MonitoringLocationIdentifier, OrganizationIdentifier) %>%
+      # determine the number of Depths per group
+      dplyr::mutate(
+        DepthsPerGroup = length(unique(Depth)),
+        # determine bottom value using TADA.ActivityBottomDepthHeightMeasure.MeasureValue or the max depth record for profile data
+        Bottom = ifelse(DepthsPerGroup > 1 & is.na(TADA.ActivityBottomDepthHeightMeasure.MeasureValue), max(Depth), TADA.ActivityBottomDepthHeightMeasure.MeasureValue)
+      ) %>%
+      dplyr::ungroup() %>%
+      # assign depth categories by using depth information
+      dplyr::mutate(TADA.DepthCategory.Flag = dplyr::case_when(
+        !MonitoringLocationTypeName %in% c("Lake, Reservoir, Impoundment", "Great Lakes", "Lake", "Great Lake") ~ "Not Calculated For MonitoringLocationType",
```

Comment:
Does this function only work for these location types?
"Lake, Reservoir, Impoundment", "Great Lakes", "Lake", "Great Lake"

Suggest to add details about this to function documentation.